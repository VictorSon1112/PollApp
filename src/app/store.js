import { configureStore } from '@reduxjs/toolkit';
import userReducer from "./userSlice";
import questionReducer from "./questionSlice";
import infoUserReducer from "./infoUserSlice";
export const store = configureStore({
  reducer: {
    infoUser: infoUserReducer,
    users: userReducer,
    question: questionReducer,
  },
});
