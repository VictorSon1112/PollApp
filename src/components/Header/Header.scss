.header-app {
    display: flex;
    align-items: center;
    gap: 20px;

    .logo {
        width: 120px;
        background: rgba(255, 255, 255, 0.2);
        color: #fff;
        font-weight: bold;
        text-transform: uppercase;
        font-size: 12px;
        border-radius: 20px;
        height: 50px;
        padding: 0;
        display: flex;
        align-items: center;
        justify-content: center;
    }

    .header-menu {
        flex: 1;
        width: 100%;
    }

    .user-info {
        display: flex;
        align-items: center;
    }

    .user-avatar {
        width: 40px;
        height: 40px;
        overflow: hidden;
        border-radius: 50%;
        margin-right: 16px;
    }

    .user-avatar img {
        width: 100%;
        height: 100%;
        object-fit: cover;
    }

    .user-name {
        color: #fff;
        font-size: 16px;
        font-weight: bold;
    }

    .logout-btn {
        width: 100px;
    }
}