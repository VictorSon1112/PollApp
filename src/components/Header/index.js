import {
  FormOutlined,
  HomeOutlined,
  LogoutOutlined,
  TrophyOutlined,
} from "@ant-design/icons";
import { Button, Layout, Menu } from "antd";
import { useNavigate } from "react-router-dom";
import "./Header.scss";
import { useDispatch } from "react-redux";
import { logout } from "../../app/infoUserSlice";

function Header({ user }) {
  const navigate = useNavigate();
  const dispatch = useDispatch();

  const handleChangeMenu = (item) => {
    navigate(`/${item.key}`);
  };

  const handleLogout = () => {
    dispatch(logout());
  };

  return (
    <Layout.Header className="header-app">
      <div className="wrap-logo" style={{ cursor: "default" }}>
        <div className="logo">Poll App</div>
      </div>
      <Menu
        theme="dark"
        mode="horizontal"
        defaultSelectedKeys={["home"]}
        className="header-menu"
        onClick={handleChangeMenu}
        items={[
          {
            key: "",
            icon: <HomeOutlined />,
            label: "Home",
          },
          {
            key: "leaderboard",
            icon: <TrophyOutlined />,
            label: "LeaderBoard",
          },
          {
            key: "add",
            icon: <FormOutlined />,
            label: "CreatePoll",
          },
        ]}
      />

      <div className="user-info">
        <div className="user-avatar">
          <img src={user.avatarURL} alt={user.name} />
        </div>
        <div className="user-name">{user.name}</div>
      </div>
      <Button
        type="primary"
        icon={<LogoutOutlined />}
        className="logout-btn"
        onClick={handleLogout}
      >
        Logout
      </Button>
    </Layout.Header>
  );
}

export default Header;
