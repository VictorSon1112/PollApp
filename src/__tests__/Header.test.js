import { render } from "@testing-library/react";
import * as React from "react";
import { Provider } from "react-redux";
import store from "../app/store";
import Header from "../components/Header";
import { configureStore, getDefaultMiddleware } from "@reduxjs/toolkit";
import userReducer from "../app/userSlice";
import questionReducer from "../app/questionSlice";
import infoUserReducer from "../app/infoUserSlice";
import { BrowserRouter } from "react-router-dom";

const props = {
  user: {
    id: "sarahedo",
    password: "123123",
    name: "Sarah Edo",
    avatarURL:
      "https://img.freepik.com/premium-vector/female-user-profile-avatar-is-woman-character-screen-saver-with-emotions_505620-617.jpg",
    answers: {
      "8xf0y6ziyjabvozdd253nd": "optionOne",
      "6ni6ok3ym7mf1p33lnez": "optionOne",
      am8ehyc8byjqgar0jgpub9: "optionTwo",
      loxhs1bqm25b708cmbf3g: "optionTwo",
    },
    questions: ["8xf0y6ziyjabvozdd253nd", "am8ehyc8byjqgar0jgpub9"],
  },
};

describe("Header component", () => {
  let store;

  beforeEach(() => {
    store = configureStore({
      reducer: {
        infoUser: infoUserReducer,
        users: userReducer,
        question: questionReducer,
      },
      middleware: [...getDefaultMiddleware()],
    });
  });

  it("should render container correctly", () => {
    const wrapper = render(
      <Provider store={store}>
        <BrowserRouter>
          <Header {...props} />
        </BrowserRouter>
      </Provider>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
