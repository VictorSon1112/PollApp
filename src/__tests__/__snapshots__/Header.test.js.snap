// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render container correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="ant-layout-header header-app css-dev-only-do-not-override-ed5zg0"
      >
        <div
          class="wrap-logo"
          style="cursor: default;"
        >
          <div
            class="logo"
          >
            Poll App
          </div>
        </div>
        <ul
          class="ant-menu-overflow ant-menu ant-menu-root ant-menu-horizontal ant-menu-dark header-menu css-dev-only-do-not-override-ed5zg0"
          data-menu-list="true"
          role="menu"
          tabindex="0"
        >
          <li
            class="ant-menu-overflow-item ant-menu-item"
            data-menu-id="rc-menu-uuid-test-"
            role="menuitem"
            style="opacity: 1; order: 0;"
            tabindex="-1"
          >
            <span
              aria-label="home"
              class="anticon anticon-home ant-menu-item-icon"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="home"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M946.5 505L560.1 118.8l-25.9-25.9a31.5 31.5 0 00-44.4 0L77.5 505a63.9 63.9 0 00-18.8 46c.4 35.2 29.7 63.3 64.9 63.3h42.5V940h691.8V614.3h43.4c17.1 0 33.2-6.7 45.3-18.8a63.6 63.6 0 0018.7-45.3c0-17-6.7-33.1-18.8-45.2zM568 868H456V664h112v204zm217.9-325.7V868H632V640c0-22.1-17.9-40-40-40H432c-22.1 0-40 17.9-40 40v228H238.1V542.3h-96l370-369.7 23.1 23.1L882 542.3h-96.1z"
                />
              </svg>
            </span>
            <span
              class="ant-menu-title-content"
            >
              Home
            </span>
          </li>
          <li
            class="ant-menu-overflow-item ant-menu-item"
            role="menuitem"
            style="opacity: 1; order: 1;"
            tabindex="-1"
          >
            <span
              aria-label="trophy"
              class="anticon anticon-trophy ant-menu-item-icon"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="trophy"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M868 160h-92v-40c0-4.4-3.6-8-8-8H256c-4.4 0-8 3.6-8 8v40h-92a44 44 0 00-44 44v148c0 81.7 60 149.6 138.2 162C265.7 630.2 359 721.7 476 734.5v105.2H280c-17.7 0-32 14.3-32 32V904c0 4.4 3.6 8 8 8h512c4.4 0 8-3.6 8-8v-32.3c0-17.7-14.3-32-32-32H548V734.5C665 721.7 758.3 630.2 773.8 514 852 501.6 912 433.7 912 352V204a44 44 0 00-44-44zM184 352V232h64v207.6a91.99 91.99 0 01-64-87.6zm520 128c0 49.1-19.1 95.4-53.9 130.1-34.8 34.8-81 53.9-130.1 53.9h-16c-49.1 0-95.4-19.1-130.1-53.9-34.8-34.8-53.9-81-53.9-130.1V184h384v296zm136-128c0 41-26.9 75.8-64 87.6V232h64v120z"
                />
              </svg>
            </span>
            <span
              class="ant-menu-title-content"
            >
              LeaderBoard
            </span>
          </li>
          <li
            class="ant-menu-overflow-item ant-menu-item"
            role="menuitem"
            style="opacity: 1; order: 2;"
            tabindex="-1"
          >
            <span
              aria-label="form"
              class="anticon anticon-form ant-menu-item-icon"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="form"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M904 512h-56c-4.4 0-8 3.6-8 8v320H184V184h320c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V520c0-4.4-3.6-8-8-8z"
                />
                <path
                  d="M355.9 534.9L354 653.8c-.1 8.9 7.1 16.2 16 16.2h.4l118-2.9c2-.1 4-.9 5.4-2.3l415.9-415c3.1-3.1 3.1-8.2 0-11.3L785.4 114.3c-1.6-1.6-3.6-2.3-5.7-2.3s-4.1.8-5.7 2.3l-415.8 415a8.3 8.3 0 00-2.3 5.6zm63.5 23.6L779.7 199l45.2 45.1-360.5 359.7-45.7 1.1.7-46.4z"
                />
              </svg>
            </span>
            <span
              class="ant-menu-title-content"
            >
              CreatePoll
            </span>
          </li>
          <li
            aria-hidden="true"
            class="ant-menu-overflow-item ant-menu-overflow-item-rest ant-menu-submenu ant-menu-submenu-horizontal"
            role="none"
            style="opacity: 0; height: 0px; overflow-y: hidden; order: 9007199254740991; pointer-events: none; position: absolute;"
          >
            <div
              aria-controls="rc-menu-uuid-test-rc-menu-more-popup"
              aria-expanded="false"
              aria-haspopup="true"
              class="ant-menu-submenu-title"
              data-menu-id="rc-menu-uuid-test-rc-menu-more"
              role="menuitem"
              tabindex="-1"
            >
              <span
                aria-label="ellipsis"
                class="anticon anticon-ellipsis"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="ellipsis"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
                  />
                </svg>
              </span>
              <i
                class="ant-menu-submenu-arrow"
              />
            </div>
          </li>
        </ul>
        <div
          aria-hidden="true"
          style="display: none;"
        />
        <div
          class="user-info"
        >
          <div
            class="user-avatar"
          >
            <img
              alt="Sarah Edo"
              src="https://img.freepik.com/premium-vector/female-user-profile-avatar-is-woman-character-screen-saver-with-emotions_505620-617.jpg"
            />
          </div>
          <div
            class="user-name"
          >
            Sarah Edo
          </div>
        </div>
        <button
          class="ant-btn css-dev-only-do-not-override-ed5zg0 ant-btn-primary logout-btn"
          type="button"
        >
          <span
            aria-label="logout"
            class="anticon anticon-logout"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="logout"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M868 732h-70.3c-4.8 0-9.3 2.1-12.3 5.8-7 8.5-14.5 16.7-22.4 24.5a353.84 353.84 0 01-112.7 75.9A352.8 352.8 0 01512.4 866c-47.9 0-94.3-9.4-137.9-27.8a353.84 353.84 0 01-112.7-75.9 353.28 353.28 0 01-76-112.5C167.3 606.2 158 559.9 158 512s9.4-94.2 27.8-137.8c17.8-42.1 43.4-80 76-112.5s70.5-58.1 112.7-75.9c43.6-18.4 90-27.8 137.9-27.8 47.9 0 94.3 9.3 137.9 27.8 42.2 17.8 80.1 43.4 112.7 75.9 7.9 7.9 15.3 16.1 22.4 24.5 3 3.7 7.6 5.8 12.3 5.8H868c6.3 0 10.2-7 6.7-12.3C798 160.5 663.8 81.6 511.3 82 271.7 82.6 79.6 277.1 82 516.4 84.4 751.9 276.2 942 512.4 942c152.1 0 285.7-78.8 362.3-197.7 3.4-5.3-.4-12.3-6.7-12.3zm88.9-226.3L815 393.7c-5.3-4.2-13-.4-13 6.3v76H488c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h314v76c0 6.7 7.8 10.5 13 6.3l141.9-112a8 8 0 000-12.6z"
              />
            </svg>
          </span>
          <span>
            Logout
          </span>
        </button>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="ant-layout-header header-app css-dev-only-do-not-override-ed5zg0"
    >
      <div
        class="wrap-logo"
        style="cursor: default;"
      >
        <div
          class="logo"
        >
          Poll App
        </div>
      </div>
      <ul
        class="ant-menu-overflow ant-menu ant-menu-root ant-menu-horizontal ant-menu-dark header-menu css-dev-only-do-not-override-ed5zg0"
        data-menu-list="true"
        role="menu"
        tabindex="0"
      >
        <li
          class="ant-menu-overflow-item ant-menu-item"
          data-menu-id="rc-menu-uuid-test-"
          role="menuitem"
          style="opacity: 1; order: 0;"
          tabindex="-1"
        >
          <span
            aria-label="home"
            class="anticon anticon-home ant-menu-item-icon"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="home"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M946.5 505L560.1 118.8l-25.9-25.9a31.5 31.5 0 00-44.4 0L77.5 505a63.9 63.9 0 00-18.8 46c.4 35.2 29.7 63.3 64.9 63.3h42.5V940h691.8V614.3h43.4c17.1 0 33.2-6.7 45.3-18.8a63.6 63.6 0 0018.7-45.3c0-17-6.7-33.1-18.8-45.2zM568 868H456V664h112v204zm217.9-325.7V868H632V640c0-22.1-17.9-40-40-40H432c-22.1 0-40 17.9-40 40v228H238.1V542.3h-96l370-369.7 23.1 23.1L882 542.3h-96.1z"
              />
            </svg>
          </span>
          <span
            class="ant-menu-title-content"
          >
            Home
          </span>
        </li>
        <li
          class="ant-menu-overflow-item ant-menu-item"
          role="menuitem"
          style="opacity: 1; order: 1;"
          tabindex="-1"
        >
          <span
            aria-label="trophy"
            class="anticon anticon-trophy ant-menu-item-icon"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="trophy"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M868 160h-92v-40c0-4.4-3.6-8-8-8H256c-4.4 0-8 3.6-8 8v40h-92a44 44 0 00-44 44v148c0 81.7 60 149.6 138.2 162C265.7 630.2 359 721.7 476 734.5v105.2H280c-17.7 0-32 14.3-32 32V904c0 4.4 3.6 8 8 8h512c4.4 0 8-3.6 8-8v-32.3c0-17.7-14.3-32-32-32H548V734.5C665 721.7 758.3 630.2 773.8 514 852 501.6 912 433.7 912 352V204a44 44 0 00-44-44zM184 352V232h64v207.6a91.99 91.99 0 01-64-87.6zm520 128c0 49.1-19.1 95.4-53.9 130.1-34.8 34.8-81 53.9-130.1 53.9h-16c-49.1 0-95.4-19.1-130.1-53.9-34.8-34.8-53.9-81-53.9-130.1V184h384v296zm136-128c0 41-26.9 75.8-64 87.6V232h64v120z"
              />
            </svg>
          </span>
          <span
            class="ant-menu-title-content"
          >
            LeaderBoard
          </span>
        </li>
        <li
          class="ant-menu-overflow-item ant-menu-item"
          role="menuitem"
          style="opacity: 1; order: 2;"
          tabindex="-1"
        >
          <span
            aria-label="form"
            class="anticon anticon-form ant-menu-item-icon"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="form"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M904 512h-56c-4.4 0-8 3.6-8 8v320H184V184h320c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V520c0-4.4-3.6-8-8-8z"
              />
              <path
                d="M355.9 534.9L354 653.8c-.1 8.9 7.1 16.2 16 16.2h.4l118-2.9c2-.1 4-.9 5.4-2.3l415.9-415c3.1-3.1 3.1-8.2 0-11.3L785.4 114.3c-1.6-1.6-3.6-2.3-5.7-2.3s-4.1.8-5.7 2.3l-415.8 415a8.3 8.3 0 00-2.3 5.6zm63.5 23.6L779.7 199l45.2 45.1-360.5 359.7-45.7 1.1.7-46.4z"
              />
            </svg>
          </span>
          <span
            class="ant-menu-title-content"
          >
            CreatePoll
          </span>
        </li>
        <li
          aria-hidden="true"
          class="ant-menu-overflow-item ant-menu-overflow-item-rest ant-menu-submenu ant-menu-submenu-horizontal"
          role="none"
          style="opacity: 0; height: 0px; overflow-y: hidden; order: 9007199254740991; pointer-events: none; position: absolute;"
        >
          <div
            aria-controls="rc-menu-uuid-test-rc-menu-more-popup"
            aria-expanded="false"
            aria-haspopup="true"
            class="ant-menu-submenu-title"
            data-menu-id="rc-menu-uuid-test-rc-menu-more"
            role="menuitem"
            tabindex="-1"
          >
            <span
              aria-label="ellipsis"
              class="anticon anticon-ellipsis"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="ellipsis"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
                />
              </svg>
            </span>
            <i
              class="ant-menu-submenu-arrow"
            />
          </div>
        </li>
      </ul>
      <div
        aria-hidden="true"
        style="display: none;"
      />
      <div
        class="user-info"
      >
        <div
          class="user-avatar"
        >
          <img
            alt="Sarah Edo"
            src="https://img.freepik.com/premium-vector/female-user-profile-avatar-is-woman-character-screen-saver-with-emotions_505620-617.jpg"
          />
        </div>
        <div
          class="user-name"
        >
          Sarah Edo
        </div>
      </div>
      <button
        class="ant-btn css-dev-only-do-not-override-ed5zg0 ant-btn-primary logout-btn"
        type="button"
      >
        <span
          aria-label="logout"
          class="anticon anticon-logout"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="logout"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M868 732h-70.3c-4.8 0-9.3 2.1-12.3 5.8-7 8.5-14.5 16.7-22.4 24.5a353.84 353.84 0 01-112.7 75.9A352.8 352.8 0 01512.4 866c-47.9 0-94.3-9.4-137.9-27.8a353.84 353.84 0 01-112.7-75.9 353.28 353.28 0 01-76-112.5C167.3 606.2 158 559.9 158 512s9.4-94.2 27.8-137.8c17.8-42.1 43.4-80 76-112.5s70.5-58.1 112.7-75.9c43.6-18.4 90-27.8 137.9-27.8 47.9 0 94.3 9.3 137.9 27.8 42.2 17.8 80.1 43.4 112.7 75.9 7.9 7.9 15.3 16.1 22.4 24.5 3 3.7 7.6 5.8 12.3 5.8H868c6.3 0 10.2-7 6.7-12.3C798 160.5 663.8 81.6 511.3 82 271.7 82.6 79.6 277.1 82 516.4 84.4 751.9 276.2 942 512.4 942c152.1 0 285.7-78.8 362.3-197.7 3.4-5.3-.4-12.3-6.7-12.3zm88.9-226.3L815 393.7c-5.3-4.2-13-.4-13 6.3v76H488c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h314v76c0 6.7 7.8 10.5 13 6.3l141.9-112a8 8 0 000-12.6z"
            />
          </svg>
        </span>
        <span>
          Logout
        </span>
      </button>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
